tests/test_loops.py

```diff
@@ -88,6 +130,45 @@ def test_single_loop(loop_args):
         lit_loop.run_single_loop(lit_api_mock, None, requests_queue, transport, callback_runner=NOOP_CB_RUNNER)
 
 
+@pytest.mark.asyncio
+async def test_single_loop_process_single_async_request(async_loop_args, mock_transport):
+    lit_api_mock, requests_queue = async_loop_args
+
+    # Get a request from the queue (already populated by the fixture)
+    request = requests_queue.get()
+    loop = SingleLoop()
+    await loop._process_single_request(
+        request,
+        lit_api_mock,
+        None,
+        mock_transport,
+        NOOP_CB_RUNNER,
+    )
+    response = await mock_transport.areceive(consumer_id=request[0])
+    expected_output = request[3]["input"] ** 2
+    assert response == (request[1], ({"output": expected_output}, ls.utils.LitAPIStatus.OK))
+
+
+def test_run_single_loop_with_async(async_loop_args, monkeypatch):
+    mock_transport = MockMPQueueTransport(num_consumers=2)
+    lit_api_mock, requests_queue = async_loop_args
+
+    loop = SingleLoop()
+
+    # Patch kill to do nothing in test
+    monkeypatch.setattr(loop, "kill", lambda: None)
```

Comment:
nice, I think we might have to look at kill later on. It has introduced some weird behavior while terminating the server. maybe only allow it for windows.

Reply:
sure.